Classes/Hooks/DefaultUploadFolder.php

```diff
@@ -140,35 +142,38 @@ protected function getDefaultUploadFolderForAllTables(
         array $userTsConfig
     ) {
         $subFolder = $defaultPageTs[self::DEFAULT_UPLOAD_FOLDERS][self::DEFAULT_FOR_ALL_TABLES] ?? '';
+
+        $dateFormatConfig = $defaultPageTs[self::DEFAULT_UPLOAD_FOLDERS][self::DEFAULT_FOR_ALL_TABLES . '.'] ?? [];
+        $subFolder = $this->checkAndConvertForDateFormat($subFolder, $dateFormatConfig);
         if (empty($subFolder)) {
             $subFolder = $userTsConfig[self::DEFAULT_UPLOAD_FOLDERS][self::DEFAULT_FOR_ALL_TABLES] ?? '';
-            $subFolder = $this->checkAndConvertForDateFormat($subFolder,$userTsConfig,self::DEFAULT_FOR_ALL_TABLES);
-        }else{
-            $subFolder = $this->checkAndConvertForDateFormat($subFolder,$defaultPageTs,self::DEFAULT_FOR_ALL_TABLES);
-        }
 
+            $dateFormatConfig = $userTsConfig[self::DEFAULT_UPLOAD_FOLDERS][self::DEFAULT_FOR_ALL_TABLES . '.'] ?? [];
+            $subFolder = $this->checkAndConvertForDateFormat($subFolder, $dateFormatConfig);
+        }
         return $subFolder;
     }
-    protected function checkAndConvertForDateFormat($subFolder, $configTs, $table){
-        $table = $table ?? 'defaultForAllTables';
-
-        if(isset($configTs['default_upload_folders.'][$table . '.'])){
-            if($configTs['default_upload_folders.'][$table.'.']['dateformat'] == 1){
-                $strplace = ['{Y}','{y}',
-                    '{m}','{n}',
-                    '{j}','{d}',
-                    '{W}','{w}'
-                ];
-                $replaceWith = [
-                    date('Y'), date('y'),
-                    date('m'), date('n'),
-                    date('j'), date('d'),
-                    date('W'), date('w')
-                ];
-                $subFolder = str_replace($strplace, $replaceWith, $subFolder);
-            }
+
+    protected function checkAndConvertForDateFormat($subFolder, $dateFormatConfig)
```

Comment:
Add phpdoc annotations for the function, a return type and types for the parameters